test/unit/puppetlabs/services/jruby/jruby_puppet_testutils.clj

```diff
@@ -181,3 +189,114 @@ create-mock-pool-instance :- JRubyInstance
     (while (< (count (jruby-core/registered-instances pool-context))
               num-jrubies)
       (Thread/sleep 100))))
+
+(schema/defn ^:always-validate
+  mock-puppet-config-settings :- {schema/Str schema/Any}
+  "Return a map of settings that mock the settings that core Ruby Puppet
+  would return via a call to JRubyPuppet.getSetting()."
+  [jruby-puppet-config :- {:master-conf-dir schema/Str
+                           :master-code-dir schema/Str
+                           schema/Keyword schema/Any}]
+  (let [certname "localhost"
+        confdir (:master-conf-dir jruby-puppet-config)
+        ssldir (str confdir "/ssl")
+        certdir (str ssldir "/certs")
+        cadir (str ssldir "/ca")
+        private-key-dir (str ssldir "/private_keys")]
+    {"allow_duplicate_certs" false
+     "autosign" true
+     "keylength" 2048
+     "cacert" (str cadir "/ca_crt.pem")
+     "ca_name" (str "Puppet CA: " certname)
+     "cacrl" (str cadir "/ca_crl.pem")
+     "cakey" (str cadir "/ca_key.pem")
+     "capub" (str cadir "/ca_pub.pem")
+     "ca_ttl" 157680000
+     "certdir" certdir
+     "certname" certname
+     "cert_inventory" (str cadir "/inventory.txt")
+     "codedir" (:master-code-dir jruby-puppet-config)
+     "csr_attributes" (str confdir "/csr_attributes.yaml")
+     "csrdir" (str cadir "/requests")
+     "data_binding_terminus" "hiera"
+     "dns_alt_names" ""
+     "hostcert" (str certdir "/" certname ".pem")
+     "hostcrl" (str ssldir "/crl.pem")
+     "hostprivkey" (str private-key-dir "/" certname ".pem")
+     "hostpubkey" (str ssldir "/public_keys/" certname ".pem")
+     "localcacert" (str certdir "/ca.pem")
+     "manage_internal_file_permissions" true
+     "privatekeydir" private-key-dir
+     "requestdir" (str ssldir "/certificate_requests")
+     "serial" (str cadir "/serial")
+     "signeddir" (str cadir "/signed")
+     "ssl_client_header" "HTTP_X_CLIENT_DN"
+     "ssl_client_verify_header" "HTTP_X_CLIENT_VERIFY"
+     "trusted_oid_mapping_file" (str confdir
+                                     "/custom_trusted_oid_mapping.yaml")}))
```

Comment:
I think one of the problems with the memoizing approach is that many of the settings are calculated based on some base settings whose values change - especially the `:master-conf-dir`.  Many tests use a "tmp" `:master-conf-dir`.  If we had to memoize the settings per `:master-conf-dir` used, might negate a lot of the savings we get by avoiding JRuby entirely.

One thing I could do to make it more obvious that a new setting needs to be stubbed into the test is have the `getSetting` method in the `create-mock-jruby-puppet` helper explicitly throw an `Exception` if the requested setting isn't found in the map.  If we did that, a bunch of tests would fail if a new setting were plumbed but no value were stubbed into this map.  I could also add a comment in the `puppet-server-config-service` which indicates that if a value is added to the whitelist that an entry needs to be inserted into this map.

Do you think that would work?